### PR TITLE
Plugin scan dialog freeze

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #3944 

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Review by Paul-Licameli — APPROVED** [review]
I think the changes in Buffered IPCChannel.cpp are unnecessary for the fix but harmless to add.  Am I right?

File: libraries/lib-ipc/internal/BufferedIPCChannel.cpp [hunk 1]
```diff
@@ -46,6 +46,10 @@ BufferedIPCChannel::~BufferedIPCChannel()
 
 void BufferedIPCChannel::Send(const void* bytes, size_t length)
 {
+   assert(length > 0);
```
[the remark this answers is not in the document]
Comment (vsverchinsky) [reply libraries/lib-ipc/internal/BufferedIPCChannel.cpp:49]: > Should this assert be here? The case is handled anyway. At line 143 of PluginHost.cpp, it appears that this is a case expected to be handled.

For debug purpose only.

**Comment by vsverchinsky** [comment]
> I think the changes in Buffered IPCChannel.cpp are unnecessary for the fix but harmless to add. Am I right?

Yes

**Comment by wpostma** [comment]
Can still repro an endless hang of plugin scan (searching for plugins) at startup with build containing commit 2c77646
[ audacity-macOS-3.2.2-beta-20221116+2c77646-universal.zip ]

**Comment by wpostma** [comment]
Is the skip button new in 3.2.2?

**Comment by wpostma** [comment]
Also get Exception Code 0x5 on my M1 Mac Studio with the above build containing 2c77646 git rev.

--
[crash.zip](https://github.com/audacity/audacity/files/10026165/crash.zip)

**Comment by wpostma** [comment]
I have a feeling if a particular plugin takes more than 1 minute to finish it should be skipped and the next plugin should be scanned, and the failed one should be blacklisted.

**Comment by wpostma** [comment]
After removing LoopMix by AudioModern from my system, the build containing 2c77646 completes without a hang.

Not sure it's a bug still. You can get past the hang by clicking skip.

**Pull request merged**